exonum-java-binding/testkit/src/main/java/com/exonum/binding/testkit/TestKit.java

```diff
@@ -97,100 +104,105 @@
   public static final short MAX_VALIDATOR_COUNT_WITH_ENABLED_TIME_SERVICE = 3;
   @VisibleForTesting
   static final short MAX_SERVICE_NUMBER = 256;
+  @VisibleForTesting
+  static final int MAX_SERVICE_INSTANCE_ID = 1023;
+  @VisibleForTesting
+  static final Any DEFAULT_CONFIGURATION = Any.getDefaultInstance();
+  private static final int DEFAULT_SERVER_PORT = 0;
   private static final Serializer<Block> BLOCK_SERIALIZER = BlockSerializer.INSTANCE;
-  private static final short TIME_SERVICE_ID = 4;
+  private final ServiceRuntime serviceRuntime;
+  private final Set<Integer> timeServiceIds;
 
-  private final Map<Short, Service> services = new HashMap<>();
   @VisibleForTesting
   final Cleaner snapshotCleaner = new Cleaner("TestKit#getSnapshot");
 
-  private TestKit(long nativeHandle, Map<Short, UserServiceAdapter> serviceAdapters) {
+  private TestKit(long nativeHandle, List<TimeServiceSpec> timeServiceSpecs,
+                  ServiceRuntime serviceRuntime) {
     super(nativeHandle, true);
-    populateServiceMap(serviceAdapters);
+    this.serviceRuntime = serviceRuntime;
+    timeServiceIds = timeServiceSpecs.stream()
+        .map(t -> t.serviceId)
+        .collect(toSet());
   }
 
-  private static TestKit newInstance(List<Class<? extends ServiceModule>> serviceModules,
+  private static TestKit newInstance(TestKitServiceInstances[] serviceInstances,
                                      EmulatedNodeType nodeType, short validatorCount,
-                                     @Nullable TimeProvider timeProvider) {
-    Injector frameworkInjector = Guice.createInjector(new TestKitFrameworkModule());
-    return newInstanceWithInjector(serviceModules, nodeType, validatorCount, timeProvider,
-        frameworkInjector);
-  }
-
-  private static TestKit newInstanceWithInjector(
-      List<Class<? extends ServiceModule>> serviceModules, EmulatedNodeType nodeType,
-      short validatorCount, @Nullable TimeProvider timeProvider, Injector frameworkInjector) {
-    Map<Short, UserServiceAdapter> serviceAdapters = toUserServiceAdapters(
-        serviceModules, frameworkInjector);
+                                     List<TimeServiceSpec> timeServiceSpecs,
+                                     Optional<Path> artifactsDirectory,
+                                     Optional<Integer> serverPort) {
+    ServiceRuntime serviceRuntime = createServiceRuntime(artifactsDirectory, serverPort);
```

Comment:
I don't think the testkit shall depend on the app and use this method made for native code (which also logs some information).

As everything needed to instantiate a runtime is already available in the core, 
create a single injector with a framework module, and invoke `injector.get(ServiceRuntimeAdapter.class)`. From here you can also get a ServiceRuntime, if needed.